src/runtime/go.mod

```diff
@@ -1,6 +1,7 @@
 module github.com/kata-containers/kata-containers/src/runtime
 
 go 1.21
+toolchain go1.23.7
```

Comment:
I'm not sure if we want this change otherwise it might force a higher version that our minimum to be used, which might not be available everywhere?

Reply:
Hmm, it looks like dependabot is requiring this, maybe as x/net v0.36.0 needs go 1.23?

Reply:
I've been playing with this earlier this week, yes, it looks like this is required to make the version jump. I tried with manually incrementing some dependency versions (`go mod edit ...`, then run `make vendor`) and the result was the same. 

Yes, moving toolchain versions up can be problematic for some distros and we may break someone.

Just one example: On Azure Linux, seems like the toolchain is being constantly moved forward: https://github.com/microsoft/azurelinux/commits/3.0/SPECS/golang/golang.spec

Reply:
Yeah - we currently have a version of 1.22.11 and we know that some of our downstream projects cannot currently use 1.23 with their distro restrictions. However this is a CVE fix, and 1.23.x is 7 months ols, so I don't want to sit on it for a long time. It's a tough balance and I think we need to come back to https://github.com/kata-containers/kata-containers/issues/9841 again

Reply:
I guess that golang 1.22 is no longer a stable release, so maybe we just need to bump to 1.23 for security fixes from now on anyway?